### ISSUE TITLE
css:full is called multiple times on npm start

### ISSUE THREAD

**Issue by illepic** [phase2/p2-theme-core#12]
When first firing up a project (fresh after `npm install`) with `npm start`, `css:full` is seen starting (and finishign) multiple times. You can see it starting 3 times below:

```
illepic@woodhouse-linux ~/dev/downfall-guild/project/build/dev/d8/themes/custom/patternlab                               [22:23:55] 
> $ npm start                                                                                                 ⬡ 6.6.0 [±master ●●●]

> PatternLabStarter@9.0.0 start /home/illepic/dev/downfall-guild/project/build/dev/d8/themes/custom/patternlab
> gulp

[22:26:09] Using gulpfile ~/dev/downfall-guild/project/build/dev/d8/themes/custom/patternlab/gulpfile.js
[22:26:09] Starting 'default'...
[22:26:09] Starting 'compile'...
[22:26:09] Starting 'clean'...
[22:26:09] Starting 'clean:js'...
[22:26:09] Starting 'clean:icons'...
[22:26:09] Starting 'clean:css'...
[22:26:09] Finished 'clean:js' after 11 ms
[22:26:09] Finished 'clean:icons' after 12 ms
[22:26:09] Finished 'clean:css' after 12 ms
[22:26:09] Finished 'clean' after 12 ms
[22:26:09] Starting 'js'...
[22:26:09] Starting 'icons'...
[22:26:09] Starting 'css:full'...
[22:26:09] Starting 'pl:full'...
[22:26:09] Starting 'inject:drupal'...
[22:26:09] Starting 'icons'...
[22:26:09] Starting 'inject:pl'...
[22:26:09] Starting 'css:full'...
[22:26:09] Starting 'css:full'...
[22:26:09] Starting 'icons'...
[22:26:09] Starting 'icons'...
[22:26:09] gulp-svgicons2svgfont: Font created
[22:26:09] gulp-svgicons2svgfont: Font created
[22:26:09] gulp-svgicons2svgfont: Font created
[22:26:09] gulp-svgicons2svgfont: Font created
[22:26:09] Finished 'js' after 388 ms
[22:26:09] Finished 'icons' after 383 ms
[22:26:09] Starting 'cssFull'...
[22:26:09] Finished 'icons' after 388 ms
[22:26:09] Starting 'cssFull'...
[22:26:09] Finished 'icons' after 402 ms
[22:26:09] Finished 'icons' after 392 ms
[22:26:09] Starting 'cssFull'...
[22:26:10] Finished 'cssFull' after 734 ms
[22:26:10] Finished 'css:full' after 1.12 s
[22:26:10] Starting 'js'...
[22:26:10] Finished 'cssFull' after 741 ms
[22:26:10] Finished 'css:full' after 1.12 s
[22:26:10] Starting 'js'...
[22:26:10] Finished 'cssFull' after 738 ms
[22:26:10] Finished 'css:full' after 1.14 s
[22:26:10] Finished 'js' after 69 ms
[22:26:10] Starting 'injectPl'...
[22:26:10] Finished 'js' after 80 ms
[22:26:10] Starting 'injectDrupal'...
Drupal 8 support not ready yet for injecting into patternlab.info.yml
[22:26:10] Finished 'injectDrupal' after 522 μs
[22:26:10] Finished 'inject:drupal' after 1.22 s
[22:26:10] gulp-inject 3 files into _00-head.twig.
[22:26:10] gulp-inject 10 files into _01-foot.twig.
[22:26:10] Finished 'injectPl' after 15 ms
[22:26:10] Finished 'inject:pl' after 1.21 s
[22:26:10] Starting 'icons'...
[22:26:10] gulp-svgicons2svgfont: Font created
[22:26:10] Finished 'icons' after 21 ms
[22:26:10] Starting 'pl:scss-to-json'...
[22:26:10] Finished 'pl:scss-to-json' after 1.67 ms
[22:26:10] Starting 'plBuild'...
configuring pattern lab...
your site has been generated...
site generation took 0.14904093742371 seconds and used 18MB of memory...
be well, do good work, and keep in touch...
[22:26:10] Finished 'plBuild' after 174 ms
[22:26:10] Finished 'pl:full' after 1.42 s
[22:26:10] Finished 'compile' after 1.43 s
[22:26:10] Starting 'serve'...
[22:26:10] Starting 'watch'...
[22:26:10] Starting 'watch:js'...
[22:26:10] Starting 'watch:icons'...
[22:26:10] Starting 'watch:css'...
[22:26:10] Starting 'watch:inject:pl'...
[22:26:10] Starting 'watch:pl:scss-to-json'...
[22:26:10] Starting 'watch:pl'...
[22:26:10] Starting 'watch:inject:drupal'...
[22:26:10] Starting 'watch:drupal'...
[BS] Access URLs:
 ----------------------------------------------------------
       Local: http://localhost:3051/pattern-lab/public/
    External: http://192.168.1.109:3051/pattern-lab/public/
 ----------------------------------------------------------
          UI: http://localhost:3002
 UI External: http://192.168.1.109:3002
 ----------------------------------------------------------
[BS] Serving files from: ./
[BS] Watching files...
```

![multiple css full running](https://dl.dropboxusercontent.com/1/view/p5psc2y5lf84vfl/Apps/Shutter/Selection_178.png)

**Comment by illepic** [comment]
Besides the main `tasks.compile.push` array containing the `css:full` task ( a task that will always run in default implementations), `css:full` is found as a dependency in `drupal.js/inject:drupal` and `pattern-lab--php-twig.js/inject:pl`.

So `css:full` will get called 3 times asynchronously on startup **IF** pl injection and drupal injection is enabled.

**Comment by EvanLovely** [comment]
Interesting. I think in Gulp 3 it would see those 3 "requests" to run the task and only run it once. Perhaps the upgrade to Gulp 4 is treating those differently. I'll look into this. A denounce would be a good idea, but I wouldn't want a delay from a watch being triggered.

**Comment by illepic** [comment]
It's good that each piece sets its dependent tasks; each gets to be "dumb" about the other tasks.

Since this is only an issue on `npm start` (the run of `default` task), I'm thinking about writing a small utility (or looking into other gulp packages) the essentially sets a "startup mode!" and checks to ensure certain tasks don't re-run until "startup mode!" is done. I'll have a PR idea up for this approach today/tomorrow.

**Comment by EvanLovely** [comment]
Sounds good. Keep in mind that the group tasks are registered in [PL starter's Gulpfile](https://github.com/phase2/pattern-lab-starter/blob/master/gulpfile.js#L26). I'm not sure if that's the best spot, perhaps it should in this repos `index.js`. I think I set it up in PL Starter so others could add tasks unique to their projects there. 

I also feel that the new `gulp.series` and `gulp.parallel` could be utilized better to help this. 

Finally, if it could be helpful, I can identify where and why we have some dependent tasks so we can rethink how to implement them better. Here's an incomplete list:
- Icons should compile before CSS because we generate a scss file that contains a sass map of the icons. It should also go before PL cause it makes a Twig file of the icons. 
- CSS & JS should compile before we inject tags into PL before PL compiles because we look to see what the compiled files names are so we can write script & link tags for them in PL; which needs to be done before PL compiles. 
- All linting should happen after the files are compiled and a refresh is triggered so we aren't waiting on lints with each save.

**Comment by illepic** [comment]
This is exactly what we're juggling: https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/#Dependency-Gotchas

Looks like our "independent modules" approach will need some re-thinking.

**Comment by EvanLovely** [comment]
Fixed in 9.0.3
